src/main/python/hydra/lib/mock_backend.py

```diff
@@ -218,7 +235,10 @@ def wait_app_ready(self, app, running_count):
                 l.info("Waiting for app [%s] to launch", app)
 
     def scale_app(self, app, scale):
-        # TODO: (AbdullahS) See if it makes sense to implement scale_app
+        l.info("Mock scale app")
```

Comment:
I would recommend deleting grouping information if a scale down of app is called. 

As some of the application ID's that are in the groups will disappear.